explorer/ast/expression.cpp

```diff
@@ -287,8 +287,11 @@ void Expression::Print(llvm::raw_ostream& out) const {
     }
     case ExpressionKind::ArrayTypeLiteral: {
       const auto& array_literal = cast<ArrayTypeLiteral>(*this);
-      out << "[" << array_literal.element_type_expression() << "; "
-          << array_literal.size_expression() << "]";
+      out << "[" << array_literal.element_type_expression() << "; ";
+      if (array_literal.has_size_expression()) {
+        out << array_literal.size_expression();
+      }
```

Comment:
```suggestion
      out << "[" << array_literal.element_type_expression() << ";";
      if (array_literal.has_size_expression()) {
        out << " " << array_literal.size_expression();
      }
```

Minor change to only print the space when there's a size expression.